### PR TITLE
Monitor JavaScript code analysis for embedded JS (YAML and HTML)

### LINKED ISSUE #3389: Monitor JavaScript code analysis for embedded JS (YAML and HTML)
Add performance monitoring metrics collections for YAML analysis

### PR DESCRIPTION
**Pull request by ilia-kebets-sonarsource**
Fixes #3389

### CODE REVIEW

**Comment by ilia-kebets-sonarsource** [comment]
As SQ will probably pick up the code duplication, there is probably an opportunity to modularize the HTML and YAML sensors into a single embedded one.

File: its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/HtmlAnalysisTest.java [hunk 1]
```diff
@@ -37,14 +40,20 @@ public class HtmlAnalysisTest {
   private static final Orchestrator orchestrator = OrchestratorStarter.ORCHESTRATOR;
 
   @Test
-  void should_raise_issues_in_html_files() {
+  void should_raise_issues_in_html_files() throws IOException {
     var projectKey = "html-project";
+    Path perfMonitoringDir = Path.of("target/monitoring/", projectKey);
```
Comment (yassin-kammoun-sonarsource) [inline its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/HtmlAnalysisTest.java:45]: We could use `var` and get rid of the `Path` import at the same time.

File: its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/YamlAnalysisTest.java [hunk 2]
```diff
@@ -37,14 +40,22 @@ public class YamlAnalysisTest {
   private static final Orchestrator orchestrator = OrchestratorStarter.ORCHESTRATOR;
 
   @Test
-  void single_line_inline_aws_lambda_for_js() {
+  void single_line_inline_aws_lambda_for_js() throws IOException {
     var projectKey = "yaml-aws-lambda-analyzed";
+
+    Path perfMonitoringDir = Path.of("target/monitoring/", projectKey);
```
Comment (yassin-kammoun-sonarsource) [inline its/plugin/tests/src/test/java/com/sonar/javascript/it/plugin/YamlAnalysisTest.java:46]: Same.

File: sonar-plugin/sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/eslint/HtmlSensorTest.java [hunk 3]
```diff
@@ -290,6 +295,39 @@ void should_not_save_cached_cpd() throws IOException {
...
+  void should_save_performance_metrics() throws Exception {
+    AnalysisResponse expectedResponse = response(
+      "{ issues: []," +
+      "\"metrics\": { \"ncloc\": [1]}, " +
+      "\"perf\":{\"parseTime\":12,\"analysisTime\":40}" +
+      "}"
+    );
+    when(bridgeServerMock.analyzeHtml(any())).thenReturn(expectedResponse);
+
+    MapSettings settings = new MapSettings();
```
Comment (yassin-kammoun-sonarsource) [inline sonar-plugin/sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/eslint/HtmlSensorTest.java:308]:
```suggestion
    var settings = new MapSettings();
```

File: sonar-plugin/sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/eslint/HtmlSensorTest.java [hunk 4]
```diff
@@ -290,6 +295,39 @@ void should_not_save_cached_cpd() throws IOException {
       .doesNotContain("Processing cache analysis of file: " + file.uri());
   }
 
+  @Test
+  void should_save_performance_metrics() throws Exception {
+    AnalysisResponse expectedResponse = response(
```
Comment (yassin-kammoun-sonarsource) [inline sonar-plugin/sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/eslint/HtmlSensorTest.java:300]:
```suggestion
    var expectedResponse = response(
```

File: sonar-plugin/sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/eslint/YamlSensorTest.java [hunk 5]
```diff
@@ -268,6 +273,39 @@ void should_not_save_cached_cpd() throws IOException {
       .doesNotContain("Processing cache analysis of file: " + file.uri());
   }
 
+  @Test
+  void should_save_performance_metrics() throws Exception {
+    AnalysisResponse expectedResponse = response(
```
Comment (yassin-kammoun-sonarsource) [inline sonar-plugin/sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/eslint/YamlSensorTest.java:278]:
```suggestion
    var expectedResponse = response(
```

File: sonar-plugin/sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/eslint/YamlSensorTest.java [hunk 6]
```diff
@@ -268,6 +273,39 @@ void should_not_save_cached_cpd() throws IOException {
...
+      "{ issues: []," +
+      "\"metrics\": { \"ncloc\": [1]}, " +
+      "\"perf\":{\"parseTime\":12,\"analysisTime\":40}" +
+      "}"
+    );
+    when(bridgeServerMock.analyzeYaml(any())).thenReturn(expectedResponse);
+
+    MapSettings settings = new MapSettings();
```
Comment (yassin-kammoun-sonarsource) [inline sonar-plugin/sonar-javascript-plugin/src/test/java/org/sonar/plugins/javascript/eslint/YamlSensorTest.java:286]:
```suggestion
    var settings = new MapSettings();
```

**Review by yassin-kammoun-sonarsource — APPROVED** [review]
LGTM!

**Review by saberduck — APPROVED** [review]
lgtm! but please look at @yassin-kammoun-sonarsource 's comments about using `var`

**Pull request merged**